File: reactivehive-core/src/test/kotlin/com/sksamuel/reactivehive/HiveTestConfig.kt
package com.sksamuel.reactivehive

import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.fs.FileSystem
import org.apache.hadoop.hive.conf.HiveConf
import org.apache.hadoop.hive.metastore.HiveMetaStoreClient

object HiveTestConfig {

  val hiveConf: HiveConf by lazy {
    HiveConf().apply {
      set("hive.metastore", "thrift")
      set("hive.metastore.uris", "thrift://localhost:9083")
    }
  }

  val client: HiveMetaStoreClient by lazy {
    HiveMetaStoreClient(hiveConf)
  }

  val conf: Configuration by lazy {
    Configuration().apply {
      set("fs.defaultFS", "hdfs://localhost:8020")
    }
  }

  val fs: FileSystem by lazy {
    FileSystem.get(conf)
  }
}

File: reactivehive-components/reactivehive-components-akkastream/src/test/scala/com/sksamuel/reactivehive/akkastream/HiveTestConfig.scala
package com.sksamuel.reactivehive.akkastream

import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.fs.FileSystem
import org.apache.hadoop.hive.conf.HiveConf
import org.apache.hadoop.hive.metastore.HiveMetaStoreClient

trait HiveTestConfig {

  implicit val hiveConf: HiveConf = new HiveConf()
  hiveConf.set("hive.metastore", "thrift")
  hiveConf.set("hive.metastore.uris", "thrift://hive-metastore:9083")

  implicit val client: HiveMetaStoreClient = new HiveMetaStoreClient(hiveConf)

  implicit val conf: Configuration = new Configuration()
  conf.set("fs.defaultFS", "hdfs://namenode:8020")

  implicit val fs: FileSystem = FileSystem.get(conf)
}
